server/game/cards/01_SOR/events/Waylay.ts

```diff
@@ -0,0 +1,25 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { EventCard } from '../../../core/card/EventCard';
+import { WildcardCardType, WildcardLocation } from '../../../core/Constants';
+
+export default class Waylay extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '7202133736',
+            internalName: 'waylay',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Return a non-leader unit to its owner\'s hand',
+            targetResolver: {
+                cardTypeFilter: WildcardCardType.NonLeaderUnit,
+                locationFilter: WildcardLocation.AnyArena,
+                immediateEffect: AbilityHelper.immediateEffects.returnToHand()
```

Comment:
Apologies, missed this earlier - could you try `returnToHandFromPlay()`? That's typically the one we want to use in this specific case. If it gives you any problems ping me in the discord

Reply:
Then you can remove the `locationFilter` because `returnToHandFromPlay` will set it automatically